app/main.py

```diff
@@ -2,0 +1,51 @@
+class Animal:
+    def __init__(self,
+                 name: str,
+                 appetite: int,
+                 is_hungry: bool = True
+                 ) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> str:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry is True:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        return 0
+
+
+class Cat(Animal):
+    def __init__(self,
+                 name: str,
+                 is_hungry: bool = True,
+                 appetite: int = 3
+                 ) -> None:
+        super().__init__(name, appetite, is_hungry)
+
+    def catch_mouse(self) -> str:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str,
+                 is_hungry: bool = True,
+                 appetite: int = 7
+                 ) -> None:
+        super().__init__(name, appetite, is_hungry)
+
+    def bring_slippers(self) -> str:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list) -> int:
+    points_sum = 0
+    for animal in animals:
+        if animal.is_hungry is True:
+            points_sum += animal.appetite
+        animal.feed()
```

Comment:
The logic here adds the animal's appetite to `points_sum` before calling `animal.feed()`. This means `points_sum` will include the appetite of animals that are fed in this iteration. If you want `points_sum` to reflect only the appetite of animals that were hungry before feeding, the current logic is correct. Otherwise, consider adjusting the logic.